helpers/compose.go

```diff
@@ -0,0 +1,166 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/compose-spec/compose-go/types"
+	"github.com/containerd/console"
+	"github.com/docker/compose-cli/api/client"
+	"github.com/docker/compose-cli/api/containers"
+	"github.com/docker/compose-cli/cli/mobycli"
+	"github.com/docker/compose-cli/utils/formatter"
+	format "github.com/docker/compose/v2/cmd/formatter"
+
+	"github.com/docker/compose/v2/pkg/api"
+	"github.com/pkg/errors"
+)
+
+type ContainerView struct {
+	ID      string
+	Image   string
+	Status  string
+	Command string
+	Ports   []string
+}
+
+func NewComposeClient(ctx context.Context) (*client.Client, error) {
+	return client.New(ctx)
+}
+
+func GetVersion(ctx context.Context, c *client.Client) (string, error) {
+	versionResult, err := mobycli.ExecSilent(ctx)
+	if versionResult == nil {
+		return "", err
+	}
+	return string(versionResult), err
+}
+
+func Up(ctx context.Context, c client.Client, projectName, composeFilePath string, detach bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	var logConsumer api.LogConsumer
+	if detach {
+		_, pipeWriter := io.Pipe()
+		logConsumer = format.NewLogConsumer(ctx, pipeWriter, false, false)
+	}
+	return c.ComposeService().Up(ctx, &project, api.UpOptions{Start: api.StartOptions{Attach: logConsumer}})
+}
+
+func Rm(ctx context.Context, c client.Client, projectName, composeFilePath string, force bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	return c.ComposeService().Remove(ctx, &project, api.RemoveOptions{Force: force})
+}
+
+func Stop(ctx context.Context, c client.Client, projectName, composeFilePath string, force bool) error {
+	if projectName == "" {
+		projectName = "meshery"
+	}
+	project := types.Project{Name: projectName, WorkingDir: filepath.Dir(composeFilePath), ComposeFiles: []string{composeFilePath}}
+
+	return c.ComposeService().Stop(ctx, &project, api.StopOptions{})
+}
+
+func Ps(ctx context.Context, c *client.Client, all, quiet bool, formatOpt string) error {
+	containerList, err := c.ContainerService().List(ctx, all)
+	if err != nil {
+		return errors.Wrap(err, "fetch containers")
```

Comment:
can we use fmt.Errorf to be in line with the other error cases in this file ?

Reply:
Done